fix: Clarify which layers are profiled in RN (#7680)
getsentry/sentry-docs: src/platforms/common/profiling/index.mdx

<Note>

> Profiling for React Native is available in alpha in SDK versions `5.8.0` and above. Currently, this supports profiling of the JavaScript layer. Combined profiling with native Android/iOS layers is not supported at this time.

Reviewer: ```suggestion
Profiling for React Native is available in alpha in SDK versions `5.8.0` and above. Currently, it supports profiling the JavaScript layer. Combined profiling with native Android/iOS layers isn't supported at this time.
```
